packages/react-tooltip/src/components/Tooltip/useTooltip.tsx

```diff
@@ -189,42 +189,38 @@ export const useTooltip_unstable = (props: TooltipProps): TooltipState => {
 
   const child = React.isValidElement(children) ? getTriggerChild(children) : undefined;
 
-  // The props to add to the trigger element (child)
-  const triggerProps: TooltipTriggerProps = {
-    onPointerEnter: useMergedEventCallbacks(child?.props?.onPointerEnter, onEnterTrigger),
-    onPointerLeave: useMergedEventCallbacks(child?.props?.onPointerLeave, onLeaveTrigger),
-    onFocus: useMergedEventCallbacks(child?.props?.onFocus, onEnterTrigger),
-    onBlur: useMergedEventCallbacks(child?.props?.onBlur, onLeaveTrigger),
-  };
-
-  // If the target prop is not provided, attach targetRef to the trigger element's ref prop
-  const childTargetRef = useMergedRefs(child?.ref, targetRef);
-  if (popperOptions.target === undefined) {
-    triggerProps.ref = childTargetRef;
-  }
+  const triggerAriaProps: Pick<TooltipTriggerProps, 'aria-label' | 'aria-labelledby' | 'aria-describedby'> = {};
 
   if (relationship === 'label') {
-    const hasLabel = child?.props && ('aria-label' in child.props || 'aria-labelledby' in child.props);
```

Comment:
We discussed this in offline and on PRs (https://github.com/microsoft/fluentui/pull/21331#discussion_r788204192), these in no sense in this check as we can't read props in certain scenarios (https://github.com/microsoft/fluentui/pull/21495#discussion_r795619711).

Reply:
Looking at props is fine. In fact, that's exactly what you're doing on line 216 with `...child?.props`.

This hasLabel check I added also makes sure that Tooltip doesn't add `aria-labelledby` if the component has `aria-label` (which does have a real effect since `aria-labelledby` takes precedence over `aria-label`).

I think the scenario you're referring to is when a component internally adds aria props in its render function. E.g.:
```tsx
const ExampleComponent: React.FC<React.HTMLAttributes<HTMLElement>> = props => {
  return <div aria-labelledby="foo" {...props} />;
};
```

In this case, Tooltip doesn't see that ExampleComponent is adding `aria-labelledby` inside its render function, so it will include `aria-label` on the component's props, _which is totally fine_. If ExampleComponent cares about that, it'll need to do the proper merging between the props it passes in and the ones it adds. It's exactly the same problem as if I had written `<ExampleComponent aria-label="bar" />` (when there's no Tooltip involved all), and it would be up to ExampleComponent to fix their prop merging bug.

Reply:
> Looking at props is fine.

No, it's not fine. It gives falsy expectation that it handles it, but it does not. You shown this in your example

These is no sense to add a check that works _sometimes_. It's much clearer to let consumers that something is already wrong.

Reply:
But your change is looking at props too. The spread `...child?.props` is looking at the props the user provided on the component's props object and make sure that the trigger doesn't override any of the existing props. There's no way to do prop merging without looking at props.

The `hasLabel` check works in exactly the same set of cases that the rest of this prop merging works.

This PR has the exact same issue for any other prop:
```tsx
const ExampleComponent: React.FC<React.HTMLAttributes<HTMLElement>> = props => {
  return <div id="foo" {...props} />;
};
```
```jsx
<MenuTrigger>
  <ExampleComponent />
</MenuTrigger>
```

In this case, MenuTrigger will set `id` on ExampleComponent because ExampleComponent's props object doesn't contain `id`. And then the `id="foo"` gets overridden by the one that MenuTrigger set.

Again, this isn't a bug on MenuTrigger: it's up to ExampleComponent to handle that, because ExampleComponent is what's doing the prop merging internally. It's the exact same problem if I were to write this, when no triggers are involved:
```jsx
<ExampleComponent id="bar" />
```

Reply:
> But your change is looking at props too. The spread `...child?.props` is looking at the props the user provided on the component's props object and make sure that the trigger doesn't override any of the existing props. There's no way to do prop merging without looking at props.
> 
> The `hasLabel` check works in exactly the same set of cases that the rest of this prop merging works.

Again, the spread `...child?.props` just spread props. If something is present in `child.props` it will override our defaults, if not then not. It does not make assumptions based on `child.props` while the check does.

Reply:
I'm trying to figure out where our disconnect is. Let me address a few of the points, and could you tell me where my thinking and your thinking diverge?

> Again, the spread `...child?.props` just spread props.

The spread is effectively looking at whether the prop exists in child.props: if so, it picks the one from child.props; if not, it picks the one from ariaProps. Do you agree these two are effectively equivalent?
```ts
// This spread:

const mergedProps = {
  ...triggerProps,
  ...child.props,
};

// Is effectively equivalent to this, even if it is a crazy way to write such a thing:

const mergedProps = {
  'aria-label': 'aria-label' in child.props ? child.props['aria-label'] : triggerProps['aria-label'],
  'aria-labelledby': 'aria-labelledby' in child.props ? child.props['aria-labelledby'] : triggerProps['aria-labelledby'],
  // ... etc. for all possible props ...
};
```

> If something is present in `child.props` it will override our defaults, if not then not.


The `hasLabel` check is very similar: the only difference is that it's checking for the existence of two props in `child.props` rather than just the one it's overriding. It's effectively the same as this:
```ts
// If the child has 'aria-label' OR 'aria-labelledby', use both 'aria-label' AND 'aria-labelledby' from the child
const hasLabel = 'aria-label' in child.props || 'aria-labelledby' in child.props;
const mergedProps = {
  'aria-label': hasLabel ? child.props['aria-label'] : triggerProps['aria-label'],
  'aria-labelledby': hasLabel ? child.props['aria-labelledby'] : triggerProps['aria-labelledby'],
  // ... combine the rest of the props normally ...
};
```

> It does not make assumptions based on `child.props` while the check does.

Could you clarify what assumptions you mean?

Reply:
> I'm trying to figure out where our disconnect is. Let me address a few of the points, and could you tell me where my thinking and your thinking diverge?

There is definitely misunderstanding 🥲 We discussed this multiple times on different PRs (https://github.com/microsoft/fluentui/pull/21331#discussion_r788204192, https://github.com/microsoft/fluentui/pull/21495#discussion_r795619711, https://github.com/microsoft/fluentui/pull/21532#discussion_r797451990) and we had multiple offline syncs about triggers.

The conclusion was AFAIR:
- Do not have such checks as they are unsafe when customers compose components. Like in the example as you provided (https://github.com/microsoft/fluentui/pull/21609#discussion_r799682401).
- Align the approach for merging props on triggers and use `...child.props` everywhere before we will figure out the safer/better pattern

Is this correct? Or I misunderstood something? As the PR implements exactly this.

> The spread is effectively looking at whether the prop exists in child.props: if so, it picks the one from child.props; if not, it picks the one from ariaProps. Do you agree these two are effectively equivalent?

If you will write a unit test for this behavior - yes, they are _equal_. But, there is a _difference_ in the approach:
- `hasLabel` checks `child.props` and has behavior changes based on `child.props`
- `...child.props` doesn't make any assumptions, it just overrides our defaults

Usage of `...child.props` also solves the problem with the example you provided. _And I agree that it's developer responsibility to fix it in such cases._ But for unknown we would like to take this responsibility on us when JSX tree matches the ideal usage 🙃 But when it does not match - it's developer's responsibility. This is something I don't understand.

The `...child.props` spread is also aligned with behavior of `resolveShorthand()`:

https://github.com/microsoft/fluentui/blob/240b1c16b5493d3d2497af761f9c179be992378f/packages/react-utilities/src/compose/resolveShorthand.ts#L33

We don't have checks for `defaultProps` in slots, right? I don't understand why it should be different triggers.

Reply:
I had also thought we agreed in our previous discussions :)

> The conclusion was AFAIR:
> 
> * Do not have such checks as they are unsafe when customers compose components. Like in the example as you provided ([fix(Popover|Tooltip): update trigger implementation to be consistent #21609 (comment)](https://github.com/microsoft/fluentui/pull/21609#discussion_r799682401)).
> * Align the approach for merging props on triggers and use `...child.props` everywhere before we will figure out the safer/better pattern
> 
> Is this correct? Or I did misunderstand something? As the PR implements exactly this.

I _agree_ we shouldn't do anything special in the case of my example you linked. In fact, the point of the example was to show that both the `'aria-label' in child.props` check and `...child.props` have the same limitation (they don't work for composed compoents), and that's fine/by design. 

The hasLabel check simply prevents us from overriding `aria-label` with `aria-labelledby` if the user included one--same as the spread preventing us from overriding `aria-label` with `aria-label` if the user provided that.

I think the disconnect is that you think of this:
```ts
if (!('aria-label' in child.props)) { 
  triggerProps['aria-labelledby'] = '...'; 
}
```
as a different (and not desirable) way of merging props compared to spreading `...ariaProps` before `...child.props`. Does that sound right?

--

The main thing the `hasLabel` check is trying to avoid is unexpected behavior in a case like this:
```jsx
<Tooltip relationship="label" content="This is a tooltip that uses aria-label">
  <Button aria-label="✅ This label wins over the tooltip's aria-label" />
</Tooltip>
```
```jsx
<Tooltip relationship="label" content={<>This is a tooltip that<br />uses aria-labelledby</>}>
  <Button aria-label="💥 This label gets overridden by the tooltip's aria-labelledby" />
</Tooltip>
```

Reply:
> The hasLabel check simply prevents us from overriding aria-label with aria-labelledby if the user included one--same as the spread preventing us from overriding aria-label with aria-label if the user provided that.

This does not work in this example (https://github.com/microsoft/fluentui/pull/21532#discussion_r797451990), right? This behavior is against "learn once". Let me quote myself:

> But for unknown reason we would like to take this responsibility on us when JSX tree matches the ideal usage... But when it does not match - it's developer's responsibility 🙃 This is something that I don't understand.
> 
> The `...child.props` spread is also aligned with behavior of `resolveShorthand()`:
> 
> https://github.com/microsoft/fluentui/blob/240b1c16b5493d3d2497af761f9c179be992378f/packages/react-utilities/src/compose/resolveShorthand.ts#L33
> 
> We don't have checks for `defaultProps` in slots, right? I don't understand why it should be different triggers.

Why we would like to have functionality that works only in some cases? And it the same time is inconsistent with other parts of library?

Reply:
> This does not work in this example ([#21532 (comment)](https://github.com/microsoft/fluentui/pull/21532#discussion_r797451990)), right? This behavior is against "learn once". 

Could you describe what you expect to happen in that case, and how it differs from how it should work? I think it does work as expected, but maybe we have different expectations.

Here's another example where it works as I would expect, using a nightly build with the `hasLabel` check in it:
https://codesandbox.io/s/aged-dream-m3xyv?file=/src/App.tsx

With your change, "Test Case 1" would have both aria-label and aria-labelledby, which overrides the aria-label set by the user. (There are more test cases in the codesandbox)

```tsx
const ButtonWithTooltip: React.FC<React.HTMLAttributes<HTMLElement>> = (props) => {
  return (
    <Tooltip content={<b>This sets aria-labelledby</b>} relationship="label">
      <button {...props} />
    </Tooltip>
  );
};
```
Test case 1:
```html
<MenuTrigger>
  <ButtonWithTooltip id="this-id-wins" aria-label="This aria-label wins, and no aria-labelledby">
    Test Case 1
  </ButtonWithTooltip>
</MenuTrigger>
```
Result:
```html
<button aria-haspopup="menu" id="this-id-wins" aria-label="This aria-label wins, and no aria-labelledby">
  Test Case 1
</button>
```

Reply:
> Could you describe what you expect to happen in that case, and how it differs from how it should work? I think it does work as expected, but maybe we have different expectations.

It should work always and prevent users from doing wrong things, but it does not (_check CodeSandbox below_). 

I modified CodeSandbox that you provided with different cases:
https://codesandbox.io/s/naughty-morning-z6rww?file=/src/App.tsx

In both customers will **silently** get `aria-label="This has both" aria-labelledby="tooltip-*"`. This shows that the check is not always safe. It does not prevent customers from doing wrong things. ➡️ There is no sense in it.

> But for unknown reason we would like to take this responsibility on us when JSX tree matches the ideal usage... But when it does not match - it's developer's responsibility 🙃 This is something that I don't understand.

For example, the approach with the warning (https://github.com/microsoft/fluentui/pull/21331#discussion_r789947542) will at least notify customers _always_.

Reply:
> I modified CodeSandbox that you provided with different cases: https://codesandbox.io/s/naughty-morning-z6rww?file=/src/App.tsx
> 
> In both customers will **silently** get `aria-label="This has both" aria-labelledby="tooltip-*"`.
> 
> This shows that the check is not always safe. It does not prevent customers from doing wrong things. ➡️ There is no sense in it.

But that's a bug with how the ButtonWithMenuTrigger component merges props. It's adding aria-label without checking what's already set on its incoming props. It would need to do its own `hasLabel`-type check to see if aria-labelledby was on props before adding its own aria-label. That bug exists with no Tooltip involved at all--just writing `<ButtonWithMenuTrigger aria-labelledby="foo" />` would result in the same behavior.

In this codesandbox, "Test Case A" shows the above bug without a tooltip involved at all. And Test Case B/C show how it could be fixed using a helper function `combineProps`. The bug, and bug fixes are all in the example components, not in Tooltip.
https://codesandbox.io/s/crazy-browser-3qcuu?file=/src/App.tsx

It might be worthwhile to have a utility function to handle merging props (like `combineProps` in that codesandbox), which handles merging between defaultProps and overrideProps. In fact, that's what I thought we had agreed on in our previous conversation :). We could use it in triggers, and all of the other places we merge props, like resolveShorthand. But until that exists, I think it's worthwhile for Tooltip to have the one-off check since it could result in [unexpected behavior](https://github.com/microsoft/fluentui/pull/21609#discussion_r799813800) otherwise.

But in any case, I think I've made my point to the best I can. If you want to remove the check that I added, you have my PR approval.

Reply:
So how I understand the discussion so far:

There are two things:
1. we merge trigger and child props
2. we conditionally apply `aria-labelledby` based on child props.

Both will break once you introduce a custom component in the middle (CITM).

To fix the problem 1, you just need to spread all props on a correct element in the CITM. This is a common requirement.
To fix the problem 2, you need to explicitly handle the `aria-labelledby` in the CITM - which is, by some, considered strange.

The `hasLabel` works in the simple use case but requires a "surprising workaround" for the CITM scenario.

The reason why we have the `hasLabel` is: _The "simple use case" is the majority of use cases, those should work out of the box._

The opinion against is: _With real world, complex applications this "majority" quickly becomes a minority. And even if that would be the 90% majority, it is better to consistently break in 100% use cases than to seamlessly work in 90% and unexpectedly fail in 10%_

---

I would personally prefer the consistency over the happy scenario simplicity.
The `hasLabel` will be fragile to break with refactoring - you have a working button with a tooltip, you refactor it to add a CITM - you test that the tooltip still works (and eventually fix the props spreading in CITM) but you will not realize you need test aria attributes as well.
Also, if we would decide to change this decision, it is easier to add `hasLabel` logic than to remove it.

---

That being said, we already has the `hasLabel` in master and that's not the main topic of this PR. Let's revert that in this PR so that we can unify the triggers and create a separate PR of RFC to continue this discussion about the `hasLabel`.